File: src/sp/it/pl/gui/objects/picker/MoodPicker.java
package sp.it.pl.gui.objects.picker;

import sp.it.pl.audio.tagging.Metadata;
import static sp.it.pl.main.App.APP;

/** Mood picker. */
public class MoodPicker extends Picker<String> {

	public MoodPicker() {
		super();
		itemSupply = () -> APP.db.getStringPool().getStrings(Metadata.Field.MOOD.name()).stream();
	}

}

File: src/sp/it/pl/gui/objects/picker/MoodPicker.kt
package sp.it.pl.gui.objects.picker

import sp.it.pl.audio.tagging.Metadata
import sp.it.pl.main.AppUtil.APP
import java.util.function.Supplier

/** Mood picker. */
class MoodPicker: Picker<String>() {
    init {
        itemSupply = Supplier { APP.db.itemUniqueValuesByField[Metadata.Field.MOOD].orEmpty().stream() }
    }
}